### ISSUE TITLE
Loading Node version of Firebase using Electron and Next.js

### ISSUE THREAD

**Issue by brotzky** [firebase/firebase-js-sdk#7639]
### Operating System

macOS 12.5.1

### Browser Version

Version 117.0.5938.62 (Official Build) (arm64)

### Firebase SDK Version

10.4.0

### Firebase SDK Product:

Auth, Firestore

### Describe your project's tooling

Next.js app loaded in Electron

### Describe the problem

When refreshing the Electron app auth is not be persisted. From what I understand, this is because Firebase is running the Node version in the electron renderer process.

Refreshing the app in browser there are no issues.

I have tried adjusting the Next.js config to alias directly to esm2017 for firebase/auth, firebase/firestore, and firebase/app. This didn't work for me.

The two videos show what I mean clearly:
Refreshing the app in browser everything is persisted
Refreshing in electron auth is run again (logo overlay)

https://github.com/firebase/firebase-js-sdk/assets/8007686/88621bbf-4e9a-4dcf-b9c3-e081aad93f58
https://github.com/firebase/firebase-js-sdk/assets/8007686/b78eeb24-e923-481f-bb39-25b345f77a59

### Steps and code to reproduce issue

- Create Next.js app that uses Firebase
- Point Electron to hosted Next.js app

```js
 const win = new BrowserWindow()
 win.loadURL('https://my-nextjs-firebase-app.com')
```

Each time the window is loaded auth is run again.

**Comment by hsubox76** [comment]
Ok, here's what I did to try to reproduce this. I created a web app that just checks if a user is logged in and signs you in if not:

```
import {initializeApp} from "firebase/app";
import {getAuth, onAuthStateChanged, signInWithEmailAndPassword} from 'firebase/auth';

const firebaseConfig = {
  ***
};

const app = initializeApp(firebaseConfig);
const auth = getAuth(app);

function writeToScreen(text) {
  console.log(text);
  document.body.innerHTML = `<p>${text}</p>`
}

onAuthStateChanged(auth, user => {
  if (!user) {
    writeToScreen('no user, signing in');
    signInWithEmailAndPassword(
      auth,
      "***",
      "***"
    );
  } else {
    writeToScreen('user: ' + user.email);
  }
});
```

I served it on localhost:8080.

I created an Electron app that loads that webapp:
```
const { app, BrowserWindow } = require('electron');
const createWindow = () => {
  const win = new BrowserWindow({
    width: 800,
    height: 600
  })

  win.loadURL('http://localhost:8080')
}
app.whenReady().then(() => {
  createWindow()
})
```

In both cases it continued to show the user's email after refreshing and even closing and opening, even when I removed the "signInWithEmailAndPassword" line. Can you modify this example in a way that causes the problem you're seeing?

There's going to be a delay in both cases before it knows if a user is signed in or not, as onAuthStateChanged is asynchronous and has a small delay whether the user is stored locally or not, but onAuthStateChanged should never return null if the user has been logged in previously.

**Comment by brotzky** [comment]
Okay, thanks for the detailed repro and steps. I'll have to do a deeper check Maybe it's not properly storing/persisting cookies/localStorage like the browser (or even indexdb).

Can you confirm when you loaded it in your repro it was loading the correct versions of Firebase in Electron?

**Comment by hsubox76** [comment]
I'm not sure what you mean by correct versions of Firebase. You mean the bundles? When I used the mjs Node file (import), it was loading node_modules/@firebase/functions/dist/esm-node/index.node.esm.js. When I used the js Node file (require) it was loading node_modules/@firebase/functions/dist/index.node.cjs.js. I don't think the bundle is affecting this issue though. Also it's going to be a totally different module resolution scheme in NextJS than raw node, so getting the correct bundle in raw Node usage isn't going to tell you much about if you''re getting the correct bundle in Next usage.
